From b0df5b9aa12e0ba28499f7833f9103ba58ca66a9 Mon Sep 17 00:00:00 2001
From: "Tong.Chen" <tong.chen@windriver.com>
Date: Fri, 12 Oct 2018 15:58:16 +0800
Subject: [PATCH 03/11] Revert "FogBugz #493306: mark the l4_sp_clk as a
 critical clock"

This reverts commit c6ec92c9222a19652ce656e9ee02e52b0a61a13b.

Get patches related with stratix10 clock manager code from sdk kernel
v4.18 to update clock manager drvier, so revert patches of old clock
manager drvier.

Signed-off-by: Tong.Chen <tong.chen@windriver.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/socfpga/clk-gate-s10.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/clk/socfpga/clk-gate-s10.c b/drivers/clk/socfpga/clk-gate-s10.c
index 55cbf5c..f0b72f1 100644
--- a/drivers/clk/socfpga/clk-gate-s10.c
+++ b/drivers/clk/socfpga/clk-gate-s10.c
@@ -25,7 +25,6 @@
 #define streq(a, b) (strcmp((a), (b)) == 0)
 
 #define SOCFPGA_CS_PDBG_CLK	"cs_pdbg_clk"
-#define SOCFPGA_L4_SP_CLK	"l4_sp_clk"
 
 #define to_socfpga_gate_clk(p) container_of(p, struct socfpga_gate_clk, hw.hw)
 
@@ -105,7 +104,7 @@ static u8 socfpga_gate_get_parent(struct clk_hw *hwclk)
 {
 	struct socfpga_gate_clk *socfpgaclk = to_socfpga_gate_clk(hwclk);
 	u32 mask;
-	u8 parent = 0;
+	u8 parent;
 
 	if (socfpgaclk->bypass_reg) {
 		mask = (0x1 << socfpgaclk->bypass_shift);
@@ -193,10 +192,7 @@ static void __init __socfpga_gate_init(struct device_node *node,
 
 	init.name = clk_name;
 	init.ops = ops;
-	if (streq(clk_name, SOCFPGA_L4_SP_CLK))
-		init.flags = CLK_IS_CRITICAL;
-	else
-		init.flags = 0;
+	init.flags = 0;
 
 	init.num_parents = of_clk_parent_fill(node, parent_names, SOCFPGA_MAX_PARENTS);
 	init.parent_names = parent_names;
-- 
1.7.9.5

